aws/sdk/integration-tests/webassembly/src/list_objects.rs

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_sdk_s3::operation::list_objects_v2::ListObjectsV2Output;
+
+async fn s3_list_objects() -> ListObjectsV2Output {
+    use crate::default_config::get_default_wasi_config;
+    use aws_sdk_s3::Client;
+
+    let shared_config = get_default_wasi_config().await;
+    let client = Client::new(&shared_config);
+    let operation = client
+        .list_objects_v2()
+        .bucket("nara-national-archives-catalog")
+        .delimiter("/")
+        .prefix("authority-records/organization/")
+        .max_keys(5);
```

Comment:
Any tests that talk to real AWS services, ideally, would live in our canary. However, we don't quite have WASM set up there 😅 

We prefer to have traffic recorded tests in our integration tests.

Do you know if there's an easy way to run WASI in a Lambda?

Reply:
The `wasmtime` crate should allow you to run a WASI binary in a Rust Lambda, although I don't have any experience using it since we are calling the binaries from typescript and python. I can look into this if its a blocker for this PR, but will need some pointers around your canary setup.

Reply:
Removed the network call in https://github.com/smithy-lang/smithy-rs/pull/3409/commits/0a87c1e4af5a41b6365830fc06301ffc134ce543, will add a full end to end test to the canary soon.